modules/apps/segments/segments-service/src/main/java/com/liferay/segments/internal/criteria/contributor/EventSegmentsCriteriaContributor.java

```diff
@@ -0,0 +1,103 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.segments.internal.criteria.contributor;
+
+import com.liferay.portal.kernel.language.Language;
+import com.liferay.portal.kernel.util.Portal;
+import com.liferay.portal.odata.entity.EntityModel;
+import com.liferay.segments.criteria.Criteria;
+import com.liferay.segments.criteria.contributor.SegmentsCriteriaContributor;
+import com.liferay.segments.field.Field;
+import com.liferay.segments.internal.odata.entity.ContextEntityModel;
+import com.liferay.segments.internal.odata.entity.EntityModelFieldMapper;
+import com.liferay.segments.internal.odata.entity.EventEntityModel;
+
+import java.util.Collections;
+import java.util.List;
+
+import javax.portlet.PortletRequest;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ReferenceCardinality;
+import org.osgi.service.component.annotations.ReferencePolicy;
+import org.osgi.service.component.annotations.ReferencePolicyOption;
+
+/**
+ * @author Eduardo García
+ */
+@Component(
+	immediate = true,
+	property = {
+		"segments.criteria.contributor.key=" + EventSegmentsCriteriaContributor.KEY,
+		"segments.criteria.contributor.model.class.name=*",
+		"segments.criteria.contributor.priority:Integer=1"
+	},
+	service = SegmentsCriteriaContributor.class
+)
+public class EventSegmentsCriteriaContributor
+	implements SegmentsCriteriaContributor {
+
+	public static final String KEY = "event";
+
+	@Override
+	public EntityModel getEntityModel() {
+		return _entityModel;
+	}
+
+	@Override
+	public String getEntityName() {
+		return ContextEntityModel.NAME;
+	}
+
+	@Override
+	public List<Field> getFields(PortletRequest portletRequest) {
+		return Collections.singletonList(
+			new Field(
+				"downloaded-document-and-media",
+				_language.get(
+					_portal.getLocale(portletRequest),
+					"downloaded-document-and-media"),
+				"complex"));
+	}
+
+	@Override
+	public String getKey() {
+		return KEY;
+	}
+
+	@Override
+	public Criteria.Type getType() {
```

Comment:
Hey @cgoncas ! this should be complex right?

Reply:
Not in this PR.